.github/actions/node-prepare/action.yml

```diff
@@ -0,0 +1,21 @@
+---
+name: 'node-prepare'
+description: 'Install node packages'
+
+runs:
+  using: 'composite'
+  steps:
+  - name: ci/cache-npm-dependencies
+    uses: actions/cache@627f0f41f6904a5b1efbaed9f96d9eb58e92e920 # v3.2.4
+    with:
+      key: npm-v1-${{ hashFiles('package-lock.json') }}
+      path: node_modules
+  - name: ci/install-npm-dependencies
+    shell: bash
+    run: |
+      NODE_ENV=development npm ci --ignore-scripts
```

Comment:
That line predates this MR, but as I didn't know the answer to this i looked at the [npm ci docs](https://docs.npmjs.com/cli/v9/commands/npm-ci). It seems to be preferrable for two reasons:
- It ensures that `package.json` and `package-lock.json` in the commit being built are aligned (whereas `npm i` would update package-lock.json, if there is a mismatch), which is a nice consistency check to have
- It purges and recreates `node_modules`. This is generally good for artifact building, to avoid having extraneous packages in there from the restored npm cache. But I see this is not a problem in this PR since the [actions/setup-node cache just refers to the global packages](https://github.com/actions/setup-node#caching-global-packages-data), and not `node_modules`